fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -532,47 +535,34 @@ def _setup_bucket_strategy(self) -> None:
         # - these are only the trainable parameters, so they should produce grads
         # - they are sorted by increasing size
         self.buckets = {}
+        self._should_bucket_grad = [False for _ in self._trainable_params]
 
-        for param in self._trainable_params:
+        for i, param in enumerate(self._trainable_params):
             device = param.device
             dst_rank = self._trainable_param_to_rank[param]
 
             if param.device not in self.buckets.keys():
-                self.buckets[param.device] = [
-                    Bucket(buffer=torch.zeros(self.buffer_max_size, dtype=param.dtype, device=device))
-                    for _ in range(dist.get_world_size(self.process_group))
```

Comment:
this could be a memory leak I think, because maybe that not all ranks would have needed a bucket..